docs/enhancements/install-sno-using-ibi.md

```diff
@@ -0,0 +1,133 @@
+order to perform SNO installation using the Image-Based installation method.
+
+## Motivation
+
+The Image-Based installation brings the following advantages:
+- Users will experience faster SNO installations
+- Increased success rate installations since we restore the state of a
+  previously working SNO setup
+
+### Goals
+
+- Provide an option in Assisted-Installer to allow a user to install a SNO with
+  the Image-Based Installation method
+
+### Non-Goals
+
+- Focus is on Assisted-Installer in SaaS environment, we don’t aim to support
+  other environments at this point
+
+## Proposal
+
+### As a user, I want to install a SNO using the IBI method
+
+#### REST API changes
+
+A new flag will be added to the cluster object [TBD], along with
+extra-parameters specific to IBI [TBD].
+
+`supported-versions` API with be amended to return only the version of OCP
+available for installation with IBI.
+
+#### Feature support
+
+A new feature support will be created for the IBI installation. 
+The feature will be gated by an OCM capability until we make it available to
+all users.
+
+#### Installation
+
+When the user will hit the installation button, the `assisted-installer` will
+restore the image seed on the disk selected for installation:
+- Install RHCOS using `coreos-installer`
+- Partition the disk to create a dedicated `/var/lib/containers`
+- Grow `/` partition
+- Mount `/`, `/boot` and `/var/lib/container`
+- Download and restore the seed on the disk using `lca-cli`
+
+[Reference](https://github.com/openshift-kni/lifecycle-agent/blob/main/ib-cli/installationiso/data/install-rhcos-and-restore-seed.sh)
+
+#### Configuration
+
+After the image seed is restored on the disk, the `assisted-installer` will
+drop the configuration in `/opt/openshift`:
+- A
+  [manifest.json](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/post-pivot-configuration.md#user-specifications)
+  with: 
+   - Hostname: will be picked from the inventory of host
+   - Cluster name: will be picked from the cluster definition 
+   - BaseDomain: will be picked from the cluster definition 
+   - ClusterID: will be picked from the cluster definition 
+   - SSHkey: will be provided by the user
+   - KubeadminPasswordHash: will be generated by Assisted-Installer and
+     provided to the user
+- Pull secret
+- assisted-controller deployment manifest, it will tell us if the installation
+  succeeded, and provide a way to get installation logs?
+- generate cryptographic materials in order to provide a kubeconfig file to the
+  user
+- SSH keys
+- Eventual extra manifests
+   - OLM operator manifests
+   - Custom manifests
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+
+## Design Details [optional]
+
+
+### Open Questions
+
+When selecting Openshift version:
+- Do we plan to provide “curated” seed images?
+- Should we allow users to install their own seed?
```

Comment:
Assuming this is the direction we go, we'll need some kind of UX for the user to tell us the image. Then we'll need to ensure we don't actually try to access it from our application as that could be a security risk.

Reply:
how useful would it be that u have to generate 1 seed to install 1 cluster? :) I assume it would be usecase only for repeatable users that run heavy automation - we can check in the telemetry maybe and talk to the relevant teams.